src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+unfortunately, even that `/search` URL path is shared amongst different flavor
+of Google Search results, with URL query parameters determining which specific
+type of search result is shown. Some of those flavors use completely different
+codebases than the traditional web search result page—for example, Image Search
+and Shopping Search are both served under the `/search` URL path with different
+query parameters, but neither of those interfaces were ready to ship their own
+service worker experience (yet).
+
+**Solution**
+
+While there are [some proposals](https://github.com/w3c/ServiceWorker/issues/1373)
+that could allow for something more powerful than URL path prefixes to determine
+service worker scopes, in the present, the Google Search team was stuck
+deploying a service worker that effectively needed to do nothing for a subset of
+the pages it controlled.
+
+To work around this, the team responsible for the Search service worker built up
+a bespoke dispatch and routing framework that could be configured to check for
+criteria like the query parameters of the client page, and use those to
+determine which specific code path to go down. Rather than hardcoding rules, the
+system was built to be flexible and allow teams that share the URL space, like
+Image Search and Shopping Search, to drop in their own service worker logic down
+the line, if they decide to implement it.
+
+{% Aside 'note' %}
+While this custom solution is internal to Google, the same general principle can
+be applied to any domain that includes a number of different logical web apps,
+all of whom live under a common URL.
+{% endAside %}
+
+### Personalized results and metrics
+
+Users can sign in to Google Search using their Google Accounts, and their search
+results experience may be customized based on their particular account data.
+Logged in users are identified by specific [browser cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies),
+which is a venerable and widely-supported standard.
+
+One downside of using browser cookies, though, is that they are not exposed
+inside of a service worker, and there is no way of automatically examining their
+values and ensuring that they have not changed due to a user logging out or
+switching accounts. (There is effort underway to
+[bring cookie access to service workers](https://developers.google.com/web/updates/2018/09/asynchronous-access-to-http-cookies#welcome_service_workers),
+but as of this writing, the approach is
+[experimental](https://developers.google.com/web/updates/2018/09/asynchronous-access-to-http-cookies#origin-trial)
+and is not widely supported.)
+
+A mismatch between the service worker's view of the current logged in user and
+the actual user logged in to the Google Search web interface could lead to
+incorrectly personalized search results or misattributed metrics and logging.
+Any of those failure scenarios would be a serious issue for the Google Search
+team.
+
+**Solution**
+
```

Comment:
Suggestion to retitle solution headers to also be level three, and to call out the solution:

Solution: send cookies using postMessage